src/rules/lowercase-name.ts

```diff
@@ -103,14 +130,25 @@ export default createRule({
   } as const,
   defaultOptions: [{ ignore: [] } as { ignore: readonly JestFunctionName[] }],
   create(context, [{ ignore }]) {
+    let counter: number = 0
     return {
+
       CallExpression(node) {
         if (!isJestFunctionWithLiteralArg(node)) {
           return;
         }
+
+        if (isTestCase || isDescribe) {
```

Comment:
This conditional will always be true, as you're not actually calling the functions 😬 

([TypeScript can actually detect this in the upcoming 3.7 release](https://github.com/microsoft/TypeScript/pull/33178) 🎉)